test/ClangImporter/indirect_fields.swift

```diff
@@ -0,0 +1,19 @@
+// RUN: %target-swift-frontend -typecheck -sdk "" -I %t -I %S/Inputs/custom-modules %s -verify
+
+import IndirectFields
+
+func build_struct(a: Int32, c: Int32, d: Int32) -> StructWithIndirectField {
+    return StructWithIndirectField(__Anonymous_field0: .init(a: a), c: c, d: d)
```

Comment:
Clearly, the best solution would be to have an `init()` per argument combination, but this can rapidly become a mess. Another approach, would be to simply use anonymous parameters in the initializer for fields corresponding to anonymous struct or union.

```swift
struct StructWithIndirectField {
    init(_ __Anonymous_field0: __Unnamed_union___Anonymous_field0, c: Int32, d: Int32)
}

StructWithIndirectField(.init(a: a), c: c, d: d)
```

This would also avoid exposing the generated field name.